CRM/Sepa/Logic/Batching.php

```diff
@@ -449,100 +493,120 @@ static function closeEnded() {
+          $reference = self::getTransactionGroupReference($creditor_id, $mode, $collection_date, $financial_type_id);
 
-        $group = civicrm_api('SepaTransactionGroup', 'create', array(
+          $group = civicrm_api('SepaTransactionGroup', 'create', array(
             'version'                 => 3,
             'reference'               => $reference,
             'type'                    => $mode,
             'collection_date'         => $collection_date,
+            'financial_type_id'       => $financial_type_id,
             'latest_submission_date'  => date('Y-m-d', strtotime("-$notice days", strtotime($collection_date))),
             'created_date'            => date('Y-m-d'),
             'status_id'               => $group_status_id_open,
             'sdd_creditor_id'         => $creditor_id,
-            ));
-        if (!empty($group['is_error'])) {
-          // TODO: Error handling
-          Civi::log()->debug("org.project60.sepa: batching:syncGroups/createGroup ".$group['error_message']);
+          ));
+          if (!empty($group['is_error'])) {
+            // TODO: Error handling
+            Civi::log()->debug("org.project60.sepa: batching:syncGroups/createGroup ".$group['error_message']);
+          }
         }
-      } else {
-        $group = civicrm_api('SepaTransactionGroup', 'getsingle', array('version' => 3, 'id' => $existing_groups[$collection_date], 'status_id' => $group_status_id_open));
-        if (!empty($group['is_error'])) {
-          // TODO: Error handling
-          Civi::log()->debug("org.project60.sepa: batching:syncGroups/getGroup ".$group['error_message']);
+        else {
+          try {
+            $group = \Civi\Api4\SepaTransactionGroup::get(TRUE)
+              ->addWhere('id', '=', $existing_groups[$collection_date][$financial_type_id ?? 0])
+              ->addWhere('status_id', '=', $group_status_id_open)
+              ->execute()
+              ->single();
+          }
+          catch (Exception $exception) {
+            // TODO: Error handling
+            Civi::log()->debug('org.project60.sepa: batching:syncGroups/getGroup ' . $exception->getMessage());
+          }
+          unset($existing_groups[$collection_date][$financial_type_id ?? 0]);
         }
-        unset($existing_groups[$collection_date]);
-      }
 
-      // now we have the right group. Prepare some parameters...
-      $group_id = $group['id'];
-      $entity_ids = array();
-      foreach ($mandates as $mandate) {
-        // remark: "mandate_entity_id" in this case means the contribution ID
-        if (empty($mandate['mandate_entity_id'])) {
-          // this shouldn't happen
-          Civi::log()->debug("org.project60.sepa: batching:syncGroups mandate with bad mandate_entity_id ignored:" . $mandate['mandate_id']);
-        } else {
-          array_push($entity_ids, $mandate['mandate_entity_id']);
+        // now we have the right group. Prepare some parameters...
+        $group_id = $group['id'];
```

Comment:
`$group` might not be defined (`catch` above).

Reply:
Moved the entire code handling `$group_id` below into the `try {}`.